libraries/AP_InertialSensor/AP_InertialSensor.cpp

```diff
@@ -2006,6 +2039,29 @@ AP_InertialSensor::Gyro_Calibration_Timing AP_InertialSensor::gyro_calibration_t
     return (Gyro_Calibration_Timing)_gyro_cal_timing.get();
 }
 
+/*
+  update IMU kill mask, used for testing IMU failover
+ */
+void AP_InertialSensor::kill_imu(uint8_t imu_idx, bool kill_it)
+{
+    if (kill_it) {
+        uint8_t new_kill_mask = imu_kill_mask | (1U<<imu_idx);
+        // don't allow the last IMU to be killed
+        bool all_dead = true;
+        for (uint8_t i=0; i<MIN(_gyro_count, _accel_count); i++) {
+            if (_use[i] && !(new_kill_mask & (1U<<i))) {
```

Comment:
I'm not sure if we shouldn't be using the use_* methods here, we could be left with a single unhealthy IMU.

Reply:
fixed